samples/conntest.cpp

```diff
@@ -114,6 +114,9 @@ void sftpcon(const std::string& url) {
 
 int main(int argc,const char** argv)
 {
+    Exiv2::XmpParser::initialize();
+    ::atexit(Exiv2::XmpParser::terminate);
```

Comment:
How about using something like `IoCloser` instead of manually registering a cleanup function via `atexit`? RAII seems cleaner to me and is probably also usefull for library consumers.

Reply:
I agree with @D4N here. I used `atexit` in the past, and I think I stopped using it because I noticed that some dynamic analizers like **valgrind** do not play well with it and it keeps detecting memory leaks. Furthermore by having a RAII class taking care of the calls to `XmpParser::initialize()` and `XmpParser::terminate` we will only need to use 1 line of code in all the applications using it (Instead of 2)

Reply:
I took the freedom to add a commit doing this change. Please, let me know if you are happy about it.